examples/complete/main.tf

```diff
@@ -10,6 +10,57 @@ module "resource_group" {
   existing_resource_group_name = var.resource_group
 }
 
+##############################################################################
+# Secrets Manager Certificate Setup
+##############################################################################
+
+module "secrets_manager" {
+  count                = var.existing_secrets_manager_crn == null ? 1 : 0
+  source               = "terraform-ibm-modules/secrets-manager/ibm"
+  version              = "1.18.14"
+  secrets_manager_name = "${var.prefix}-secrets-manager"
+  sm_service_plan      = "trial"
+  resource_group_id    = module.resource_group.resource_group_id
+  region               = var.region
+}
+
+module "sm_crn" {
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.existing_secrets_manager_crn == null ? module.secrets_manager[0].secrets_manager_crn : var.existing_secrets_manager_crn
+}
+
+module "secrets_manager_private_cert_engine" {
+  source                    = "terraform-ibm-modules/secrets-manager-private-cert-engine/ibm"
+  version                   = "1.3.4"
+  secrets_manager_guid      = module.sm_crn.service_instance
+  region                    = var.region
+  root_ca_name              = "${var.prefix}-ca"
+  root_ca_common_name       = "${var.prefix}.${var.domain_name}"
+  intermediate_ca_name      = "${var.prefix}-int-ca"
+  certificate_template_name = "${var.prefix}-template"
+  root_ca_max_ttl           = "8760h"
+}
+
+module "secrets_manager_cert" {
+  # explicit depends on because the cert engine must complete creating the template before the cert is created
+  # no outputs from the private cert engine to reference in this module call
+  depends_on             = [module.secrets_manager_private_cert_engine]
+  source                 = "terraform-ibm-modules/secrets-manager-private-cert/ibm"
+  version                = "1.3.2"
+  secrets_manager_guid   = module.sm_crn.service_instance
+  secrets_manager_region = module.sm_crn.region
+  cert_name              = "${var.prefix}-kmip-cert"
+  cert_common_name       = "${var.prefix}.${var.domain_name}"
+  cert_template          = "${var.prefix}-template"
```

Comment:
for this to work with existing instance that has private cert engine already installed, you will need to expose the `cert_template` as an input to the example, and use our value from here: https://github.com/terraform-ibm-modules/common-dev-assets/blob/45fc7ba3ee51b94ce53e30ac8af14dede0919f35/common-go-assets/common-permanent-resources.yaml#L7